### DISCUSSION TITLE
Improve interpolated string handler compilation errors

### DISCUSSION THREAD

**Discussion by aelij** [dotnet/csharplang#6852] [Language Ideas] [answered]
Currently when there are no overloads that accept a `string`, the errors the compiler produces when passing a non-interpolated string to an interpolated string handler are not very clear:

```cs
void ByVal(DefaultInterpolatedStringHandler s) {}
void ByRef(ref DefaultInterpolatedStringHandler s) {}
```

Would produce these errors, respectively:
```
CS1503: Argument 1: cannot convert from 'string' to 'System.Runtime.CompilerServices.DefaultInterpolatedStringHandler'
CS1620: Argument 1 must be passed with the 'ref' keyword
```

I'd prefer something like "Expected an interpolated string".
Or alternatively, just accept a non-interpolated string as a valid value (by calling the handler's `AppendLiteral`).

**Comment by 333fred** [comment, marked as answer]
@aelij dotnet/roslyn is the place for these suggestions. You can use the bug report issue template.

**Reply by aelij** [reply]
Thanks, will do. This didn't really fit into the description of a bug report :)
